enhancements/operator-sdk/downstream-operator-sdk.md

```diff
@@ -0,0 +1,366 @@
+1. For fixes to mainline code, fix the mainline code directly. NOTE: these changes
+   will be lost when re-running UPSTREAM-MERGE.sh script.
+
+For either of the above, we will always submit the patches to the appropriate
+upstream repos.
+
+#### Directory Structures
+
+The following is a list of directories and files that we will add to the repos:
+
+* UPSTREAM-MERGE.sh # script used to sync the upstream to the downstream
+* ci
+  * dockerfiles
+    * variety of Dockerfiles used to build images for running tests
+* tests
+  * e2e test scripts
+* prow.Makefile # Makefile used by CI
+* patches
+  * contain the patch files that may be carried
+* release
+  * ansible
+    * ansible_collections # contains the sync ansible collections
+    * bin # entrypoint and other binaries for ansible base images
+    * Dockerfiles # various dockerfiles
+  * helm
+    * bin # entrypoint and other binaries for helm base images
+    * Dockerfiles # various dockerfiles
+  * sdk
+    * bin # entrypoint and other binaries for sdk base images
+    * Dockerfiles # various dockerfiles
+  * scorecard-test
+    * bin # entrypoint and other binaries for images
+    * Dockerfiles # various dockerfiles
+  * scorecard-test-kuttl
+    * bin # entrypoint and other binaries for images
+    * Dockerfiles # various dockerfiles
+* vendor
+  * contains the dependencies for the project
+
+
+### Risks and Mitigations
+
+Overall, this is a low risk enhancement since we already have existing
+processes for handling downstream builds.
+
+## Design Details
+
+### Open Questions [optional]
+
+1. Do we deliver operator-sdk on the developers.redhat.com like we do odo?
+   * https://developers.redhat.com/products/odo/overview
+
+2. Do we need an RPM?
+
+### Test Plan
+
+The existing set of Operator SDK tests will be run downstream. Today we run
+tests for the Ansible and Helm operators.
+
+### Graduation Criteria
+
+Operator SDK will go out once the downstream builds work.
+
+### Upgrade / Downgrade Strategy
+
+All Operator SDK releases have a migration guide on what users have to do to
+upgrade their operators. The downstream repos will not have upgrades or downgrades.
+
+### Version Skew Strategy
+
+* The operator-sdk repo will vendor its dependencies to keep consistent builds
+* The ansible and helm repos will also vendor any Go dependencies
+* The operator-lib repo will just be a mirror of the upstream
+
+#### Branching
+
+The downstream repos will use the standard DPTP/ART branching:
+
+* master
+* release-4.8
+* release-4.7
+* release-4.6
+
+Each of these branches will be fed by a specific upstream version stream. For
+example, for release-4.6 we are delivering v0.19.x of the Operator SDK. For
+release-4.7, the plan is to ship v1.y.z of the Operator SDK.
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in
+`Implementation History`.
+
+## Drawbacks
+
+The biggest drawback is dealing with merge conflicts from upstream repos.
```

Comment:
as long as we aren't going across versions in branches (ie, master goes from v1.0.x to v1.1.x) there shouldn't be many conflicts, at least in the past this has been pretty much the only part that produces conflicts

Reply:
Yeah I was thinking we'd follow the exact same thing where master downstream was the latest release we plan on doing. I'll redo that.